### PR TITLE
API-1366: API reference cheat sheet for product updated event

### PR DESCRIPTION
**Pull request by tomglvng**
Adding the “product updated” part first.

Peek 21-12-2020 15-43

### CODE REVIEW

File: content/events-reference/events-reference-serenity/resources/product-updated.yml [hunk 1]
```diff
@@ -0,0 +1,132 @@
+type: object
+properties:
+  events:
+    type: array
+    items:
+      type: object
+      properties:
+        action:
+          type: string
+          description: Action that triggered the event
```
Comment (meganelepalud-akeneo) [inline content/events-reference/events-reference-serenity/resources/product-updated.yml:10]:
```suggestion
          description: Event type
```

File: content/events-reference/events-reference-serenity/resources/product-updated.yml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        event_id:
+          type: string
+          description: Event identifier
```
Comment (meganelepalud-akeneo) [inline content/events-reference/events-reference-serenity/resources/product-updated.yml:13]:
```suggestion
          description: Unique event identifier
```

File: content/events-reference/events-reference-serenity/resources/product-updated.yml [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+          description: Event datetime
+        author:
+          type: string
+          description: Author of the event
+        author_type:
+          type: string
+          description: Type of the author "ui" or "api"
+        pim_source:
+          type: string
+          description: Pim source
```
Comment (meganelepalud-akeneo) [inline content/events-reference/events-reference-serenity/resources/product-updated.yml:25]:
```suggestion
          description: URL of the PIM that sends the event
```

File: content/events-reference/events-reference-serenity/resources/product-updated.yml [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+              items:
+                type: string
+            groups:
+              type: array
+              description: Codes of the groups to which the product belong
+              items:
+                type: string
+            parent:
+              type: string
+              description: Code of the parent product model when the product is a variant (only available since the 2.0). This parent can be modified since the 2.3.
```
Comment (meganelepalud-akeneo) [inline content/events-reference/events-reference-serenity/resources/product-updated.yml:50]:
```suggestion
              description: Code of the parent product model when the product is a variant
```

File: content/events-reference/events-reference-serenity/resources/product-updated.yml [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+                        description: Channel code of the product value
+                      locale:
+                        type: string
+                        description: Locale code of the product value
+                      data:
+                        type: object
+                        description: Product value
+                      linked_data:
+                        type: object
+                        description: Object containing labels of attribute options (only available in Serenity and when query parameter "with_attribute_options" is set to "true"). See <a href='/concepts/products.html#the-linked_data-format'>the `linked_data` format</a> section for more details.
```
Comment (meganelepalud-akeneo) [inline content/events-reference/events-reference-serenity/resources/product-updated.yml:70]:
```suggestion
                        description: Object containing labels of attribute options. See <a href='/concepts/products.html#the-linked_data-format'>the `linked_data` format</a> section for more details.
```

File: content/events-reference/events-reference-serenity/resources/product-updated.yml [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+                      scope:
+                        type: string
+                        description: Channel code of the product value
+                      locale:
+                        type: string
+                        description: Locale code of the product value
+                      data:
+                        type: object
+                        description: Product value
+                      linked_data:
```
Comment (meganelepalud-akeneo) [inline content/events-reference/events-reference-serenity/resources/product-updated.yml:68]: We have the linked_data by default in the event payload?

File: content/events-reference/events-reference-serenity/resources/product-updated.yml [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+                        description: Object containing labels of attribute options (only available in Serenity and when query parameter "with_attribute_options" is set to "true"). See <a href='/concepts/products.html#the-linked_data-format'>the `linked_data` format</a> section for more details.
+            associations:
+              type: object
+              properties:
+                associationTypeCode:
+                  type: object
+                  properties:
+                    groups:
+                      type: array
+                      description: Array of groups codes with which the product is in relation
```
Comment (meganelepalud-akeneo) [inline content/events-reference/events-reference-serenity/resources/product-updated.yml:79]:
```suggestion
                      description: Array of group codes with which the product is in relation
```

File: content/events-reference/events-reference-serenity/resources/product-updated.yml [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+                      items:
+                        type: string
+                    products:
+                      type: array
+                      description: Array of product identifiers with which the product is in relation
+                      items:
+                        type: string
+                    product_models:
+                      type: array
+                      description: Array of product model codes with which the product is in relation (only available since the v2.1)
```
Comment (meganelepalud-akeneo) [inline content/events-reference/events-reference-serenity/resources/product-updated.yml:89]:
```suggestion
                      description: Array of product model codes with which the product is in relation.
```

File: content/events-reference/events-reference-serenity/resources/product-updated.yml [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+                      items:
+                        type: object
+                        properties:
+                          code:
+                            type: string
+                          quantity:
+                            type: integer
+            created:
+              type: string
+              description: Date of creation
```
Comment (meganelepalud-akeneo) [inline content/events-reference/events-reference-serenity/resources/product-updated.yml:120]:
```suggestion
              description: Product creation date
```

File: content/events-reference/events-reference-serenity/resources/product-updated.yml [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+              format: dateTime
+            updated:
+              type: string
+              description: Date of the last update
```
Comment (meganelepalud-akeneo) [inline content/events-reference/events-reference-serenity/resources/product-updated.yml:124]:
```suggestion
              description: Date of the last product update
```

File: content/events-reference/events-reference-serenity/resources/product-updated.yml [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+              format: dateTime
+            metadata:
+              type: object
+              description: More information around the product (only available since the v2.0 in the Enterprise Edition)
```
Comment (meganelepalud-akeneo) [inline content/events-reference/events-reference-serenity/resources/product-updated.yml:128]:
```suggestion
              description: More information around the product (only available in the Enterprise Edition)
```

File: content/events-reference/events-reference-serenity/resources/product-updated.yml [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+              properties:
+                workflow_status:
+                  description: Status of the product regarding the user permissions (only available since the v2.0 in the Enterprise Edition)
```
Comment (meganelepalud-akeneo) [inline content/events-reference/events-reference-serenity/resources/product-updated.yml:131]:
```suggestion
                  description: Status of the product regarding the user permissions (only available in the Enterprise Edition)
```

File: content/events-reference/events-reference-v5/products-50.md [hunk 13]
```diff
@@ -1 +1,125 @@
...
+### Example
+```json
+{
+    "events": [
+        {
+            "action": "product.updated",
+            "event_id": "c306e088-fb76-479c-bbc0-18fef19da75d",
+            "event_date": "2020-10-20T09:13:59+00:00",
+            "author": "peter",
+            "pim_source": "https://pimci-pr-9739.dev.cloud.akeneo.com",
```
Comment (meganelepalud-akeneo) [inline content/events-reference/events-reference-v5/products-50.md:23]:
```suggestion
            "pim_source": "https://demo.akeneo.com",
```

File: content/events-reference/events-reference-v5/products-50.md [hunk 14]
```diff
@@ -1 +1,125 @@
...
+
+### Example
+```json
+{
+    "events": [
+        {
+            "action": "product.updated",
+            "event_id": "c306e088-fb76-479c-bbc0-18fef19da75d",
+            "event_date": "2020-10-20T09:13:59+00:00",
+            "author": "peter",
```
Comment (meganelepalud-akeneo) [inline content/events-reference/events-reference-v5/products-50.md:22]:
```suggestion
            "author": "peter",
            "author_type": "ui",
```

File: content/events-reference/events-reference-v5/products-50.md [hunk 15]
```diff
@@ -1 +1,125 @@
...
+                        "supplier_zaro"
+                    ],
+                    "enabled": true,
+                    "values": {
+                        "collection": [
+                            {
+                                "locale": null,
+                                "scope": null,
+                                "data": [
+                                    "summer_2017"
```
Comment (meganelepalud-akeneo) [inline content/events-reference/events-reference-v5/products-50.md:41]:
```suggestion
                                    "winter_2020"
```

File: content/events-reference/events-reference-v5/products-50.md [hunk 16]
```diff
@@ -1 +1,125 @@
...
+                            }
+                        ],
+                        "image": [
+                            {
+                                "locale": null,
+                                "scope": null,
+                                "data": "9/9/c/c/99cc83f105199c667505cfa8ec1458c8be4f0814_sunglasses.jpg",
+                                "_links": {
+                                    "download": {
+                                        "href": "http://localhost/api/rest/v1/media-files/9/9/c/c/99cc83f105199c667505cfa8ec1458c8be4f0814_sunglasses.jpg/download"
```
Comment (meganelepalud-akeneo) [inline content/events-reference/events-reference-v5/products-50.md:52]:
```suggestion
                                        "href": "http://demo.akeneo.com/api/rest/v1/media-files/9/9/c/c/99cc83f105199c667505cfa8ec1458c8be4f0814_sunglasses.jpg/download"
```

**Review by meganelepalud-akeneo — CHANGES_REQUESTED** [review]

File: content/events-reference/events-reference-serenity/resources/product-updated.yml (hunk 6 above)
Comment (tomglvng) [reply content/events-reference/events-reference-serenity/resources/product-updated.yml:68]: No. I will remove it from the event api doc

**Review by meganelepalud-akeneo — APPROVED** [review]

**Pull request merged**